include/mbgl/util/interpolate.hpp

```diff
@@ -143,6 +147,27 @@ struct Interpolator<Padding> {
     }
 };
 
+template <>
+struct Interpolator<VariableAnchorOffsetCollection> {
+public:
+    VariableAnchorOffsetCollection operator()(const VariableAnchorOffsetCollection& a,
+                                              const VariableAnchorOffsetCollection& b,
+                                              const float t) const noexcept {
+        assert(a.size() == b.size());
+        std::vector<AnchorOffsetPair> offsetMap;
+        for (size_t index = 0; index < a.size(); index++) {
+            const auto& aPair = a[index];
+            const auto& bPair = b[index];
+            assert(aPair.anchorType == bPair.anchorType);
+            auto offset = std::array<float, 2>{interpolate(aPair.offset[0], bPair.offset[0], t),
+                                               interpolate(aPair.offset[1], bPair.offset[1], t)};
+            offsetMap.emplace_back(aPair.anchorType, offset);
```

Comment:
This can throw which would crash the program (because of `noexcept`). Is it possible to avoid allocation?

Reply:
Uhm, I also didn't like the idea of vector in this type, but I don't know how to avoid it. It's a collection without static size limit, so we have to allocate it on heap. 

I guess it's possible to do some shenanigans with inline storage for cases when there are only few elements. But this wouldn't fix the issue, it will just make it less probable. We are open to try ideas if you have some!

Reply:
If it is unavoidable we should remove `noexcept` here (and from the base class), otherwise the program will crash when an exception is thrown here.

Reply:
Add runtime error as JS